### DISCUSSION TITLE
Named destinations

### DISCUSSION THREAD

**Discussion by chapmanwilliam** [pymupdf/PyMuPDF#3168] [Looking for help]
I wonder if anyone could point me to a ‘getting started’ place in the pymudf code for adding a functionality to add named destinations to pdf.  I can then start adding it to my fork to see how it goes.

**Comment by JorjMcKie** [comment]
There currently is no support to **_create_** named destinations.
If you want to do that, you can use low-level methods `get_new_xref()`, `xref_set_key()`, `update_object()` etc. to create required objects and their inter-relationships.
This is a complex undertaking which requires in-depth knowledge of the PDF structure, specifically name trees in the PDF catalog.
You definitely should consult PDF specification manuals in this regard.

**Reply by chapmanwilliam** [reply]
Thanks Jorji
